### PR TITLE
Adding Time Rotation Policy

### PR DESCRIPTION
**Pull request by evidanary**
This is my first time contributing to an open source project. Please do let me know if something seems off.

### CODE REVIEW

File: src/main/java/org/apache/storm/hdfs/bolt/rotation/FileTimeRotationPolicy.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+
+    public FileTimeRotationPolicy(float count, Units units){
+        this.maxMilliSeconds = (long)(count * units.getMilliSeconds());
+    }
+
+    @Override
+    public boolean mark(Tuple tuple, long offset) {
+        // The offsett is not used here as we are rotating based on time
+        long diff = (new Date()).getTime() - this.lastCheckpoint;
```
Comment (aperepel) [inline src/main/java/org/apache/storm/hdfs/bolt/rotation/FileTimeRotationPolicy.java:54]: System.currentTimeMillis() ? No need to create more Date objects

File: src/main/java/org/apache/storm/hdfs/bolt/rotation/FileTimeRotationPolicy.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return diff >= this.maxMilliSeconds;
+    }
+
+    @Override
+    public void reset() {
+        this.lastCheckpoint =  new Long((new Date()).getTime());
```
Comment (aperepel) [inline src/main/java/org/apache/storm/hdfs/bolt/rotation/FileTimeRotationPolicy.java:60]: System.currentTimeMillis() ? No need to create more Date objects

File: src/main/java/org/apache/storm/hdfs/bolt/rotation/FileTimeRotationPolicy.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            this.milliSeconds = milliSeconds;
+        }
+
+        public long getMilliSeconds(){
+            return milliSeconds;
+        }
+    }
+
+    private long maxMilliSeconds;
+    private long lastCheckpoint = new Long((new Date()).getTime());
```
Comment (aperepel) [inline src/main/java/org/apache/storm/hdfs/bolt/rotation/FileTimeRotationPolicy.java:44]: System.currentTimeMillis() ? No need to create more Date objects

**Comment by ptgoetz** [comment]
@yash-ranadive Thank you for the contribution! What you implemented was exactly how I was initially going to do it.

However, after giving it some thought, there's a problem with how `FileRotationPolicy`s work with respect to time. If you have a slow data source (i.e. you only get tuples every now and then), the file rotation may not occur at the intended time (since it is triggered by the arrival of a tuple).

So I implemented it a little bit differently, in a way that doesn't rely on the arrival of a tuple to trigger the rotation.

I'm going to close this PR, but thank you for taking the time to contribute!

**Comment by evidanary** [comment]
Great! How did you implement it? When can we see it merged?

**Comment by ptgoetz** [comment]
With a dirty use of instanceof. ;)

I also added some synchronization since the rotation could be triggered in a different thread.

Check the latest commit.

**Pull request closed without merging**